### PR TITLE
Migrate to simple reference selection and resolution

### PR DESCRIPTION
**Pull request by negz**
### Description of your changes

I've successfully tested all references by applying the updated cache, compute, and database examples included in this PR.

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [ ] Updated any relevant [documentation], [examples], or [release notes].
- [ ] Updated the dependencies in [`app.yaml`] to include any new role permissions.

[documentation]: https://github.com/crossplane/crossplane/tree/master/docs
[examples]: https://github.com/crossplane/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplane/crossplane/tree/master/PendingReleaseNotes.md
[`app.yaml`]: https://github.com/crossplane/provider-aws/blob/master/config/stack/manifests/app.yaml

### CODE REVIEW

**Comment by negz** [comment]
Okay, this is ready for review. I've successfully tested all references by applying the updated cache, compute, and database examples included in this PR.

File: apis/cache/v1beta1/replication_group_types.go [hunk 1]
```diff
@@ -374,7 +320,13 @@ type ReplicationGroupParameters struct {
 	// the CacheSecurityGroupNames.
 	// +immutable
 	// +optional
-	CacheSecurityGroupNameRefs []*SecurityGroupNameReferencerForReplicationGroup `json:"cacheSecurityGroupNameRefs,omitempty"`
+	CacheSecurityGroupNameRefs []runtimev1alpha1.Reference `json:"cacheSecurityGroupNameRefs,omitempty"`
+
+	// CacheSecurityGroupNameRefs are references to SecurityGroups used to set
+	// the CacheSecurityGroupNames.
```
Comment (hasheddan) [inline apis/cache/v1beta1/replication_group_types.go:326]: Nitpick: update comment fo `CacheSecurityGroupNameSelector`

File: apis/network/v1alpha3/internetgateway_types.go [hunk 2]
```diff
@@ -22,39 +22,19 @@ import (
...
 type InternetGatewayParameters struct {
 	// VPCID is the ID of the VPC.
 	VPCID string `json:"vpcId,omitempty"`
 
-	// VPCIDRef references to a VPC to and retrieves its vpcId
-	VPCIDRef *VPCIDReferencerForInternetGateway `json:"vpcIdRef,omitempty" resource:"attributereferencer"`
+	// VPCIDRef references a VPC to and retrieves its vpcId
+	VPCIDRef *runtimev1alpha1.Reference `json:"vpcIdRef,omitempty"`
+
+	// VPCIDSelector selects areference to a VPC to and retrieves its vpcId
```
Comment (hasheddan) [inline apis/network/v1alpha3/internetgateway_types.go:36]:
```suggestion
	// VPCIDSelector selects a reference to a VPC to and retrieves its vpcId
```

**Comment by negz** [comment]
@hasheddan I believe I've addressed your concerns - PTAL!

**Pull request merged**